capsules/extra/src/eui64.rs

```diff
@@ -0,0 +1,41 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! Provides an EUI-64 (Extended Unique Identifier) interface for userspace.
+
+use capsules_core::driver;
+pub const DRIVER_NUM: usize = driver::NUM::Eui64 as usize;
+
+use kernel::syscall::{CommandReturn, SyscallDriver};
+use kernel::{ErrorCode, ProcessId};
+
+pub struct Eui64 {
+    eui64: &'static u64,
```

Comment:
Does just
```suggestion
    eui64: u64,
```
work?

Reply:
Tyler's been working on too complex of code. He's forgotten about simple code.

Reply:
I wanted to explicitly prevent and call out that this value should not be changed by the capsule. Having just a `u64` would allow a future implementation to potentially modify this. 

I'm not beholden to `&'static` so long as our design enforces this value not being changed by the capsule.

Reply:
I don't really know how to enforce that, at the end of the day the capsule can return `CommandReturn::success_u64()` with any value it wants.

Using just a `u64` will mean the capsule has a local copy its free to change as it wants, but again, it can return whatever it wants to userspace.

Reply:
That's a fair point with the capsule potentially returning whatever it wants. We can simplify this and just use a `u64`. Perhaps @brghena is onto something with forgetting about simple code :)